### DISCUSSION TITLE
Limiting case 2 diagram

### DISCUSSION THREAD

**Discussion by cl804** [Part-II-Computational-Physics/.github#66] [General]
![image](https://user-images.githubusercontent.com/125688188/224081860-e0d78c30-2750-440f-b9bc-f07d5031bf27.png)

looks like my graph is not correct I have checked my compute magnetisation and evolve function they seem to all work. Can,t find where the issue is.

def attempt_spin_flip(lattice,width,betaJ,betaMuH):
    '''Applies the Metropolis-Hastings algorithm to try and flip a spin.'''
    i, j = np.random.randint(width, size=2)
    deltaE = compute_betaDeltaE(i, j, lattice, width, betaJ, betaMuH)
    if deltaE <= 0 or np.random.random() < np.exp(-deltaE):
        lattice[i, j] *= -1

def evolve_and_compute_M(lattice,betaJ,betaMuH,avg_times):
    '''Evolves the lattice using the Metropolis-Hastings algorithm and returns the average magnetisation computed using different time steps.'''
    average_M = []
    for t in range(avg_times[-1]+1):
        attempt_spin_flip(lattice, lattice.shape[0], betaJ, betaMuH)
        if t in avg_times:
            M = compute_magnetisation(lattice) / lattice.size
            average_M.append(M)
            M_aver = np.mean(average_M)
    return M_aver

# TO DO
betaJ_range = np.linspace(-3, 3, 100)
Ms = []

# Simulate the system at different coupling strengths
for betaJ in betaJ_range:
    # Initialize the lattice with an expected final state
    betaJ_c = (np.log(1+np.sqrt(2)))/2
    if betaJ < betaJ_c:
        lattice = init_lattice(20, 1)
    else:
        lattice = np.ones((20, 20), dtype=int)
        lattice[::2, ::2] = -1
        lattice[1::2, 1::2] = -1
    lattice = init_lattice(20, 0)
    # Evolve the system and compute the average magnetisation
    avg_times = [1000, 2000, 5000]
    M = evolve_and_compute_M(lattice, -betaJ, 0, avg_times)
    Ms.append(M)

Ma=[] 
for betaJ in betaJ_range:
    betaJ_c = (np.log(1+np.sqrt(2)))/2
    if betaJ < betaJ_c:
        M = 0
    else:
        M = np.power(1 - np.power(np.sinh(2 * betaJ),-4), 1/8)
    Ma.append(M)
# Plot the results
plt.plot(betaJ_range, Ms, label='Simulation')
plt.plot(betaJ_range, Ma, label='Analytical')
plt.xlabel ('Coupling strength betaJ')
plt.ylabel ('Magnetisation M')
plt.legend(loc='upper left')
plt.show()

here is my relevant code

**Comment by cl804** [comment]
I am not sure whether it is because the average time is not big enough
avg_times = [1000, 2000, 5000]
Usually what range of time step do we use?

**Reply by benpymer** [reply]
I went for times from 1000 up to 10000 in time steps of 100 and it seemed to work pretty well for me, but I think you can probably get away with fewer values than that.
